### PR TITLE
Reducing Code Duplication

### PR DESCRIPTION
**Pull request by LucasRyall**
# Description

Reducing the code duplication, mostly in the screens, small change to dialogue box service and animal selection display. Most of the files have the exact same changes (apologies to reviewers for there being so many files).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Testing each part that I change hasn't affected the game, and that no unit tests were affected.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/services/DialogueBoxService.java [hunk 1]
```diff
@@ -85,37 +85,32 @@ public void hideCurrentOverlay() {
...
-        if (currentEntity != null) {
-            AnimationRenderComponent animator =  currentEntity.getComponent(AnimationRenderComponent.class);
-            if (animator != null) {
-                animator.startAnimation("float");
-            }
-            currentEntity = null;
-        }
-    }
+        updateTextHelper(text, priority);
+        if (currentEntity != null) currentEntity = null; // if probably isn't necessary, but that is how it
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/services/DialogueBoxService.java:89]: Is SonarLint/Cloud happy with not having braces here?

**Review by mitsudz — APPROVED** [review]
Looks good to me - single comment below. Rudimentary play-testing showed no changes in functionality as expected. Should probably get someone from mini-game team to confirm this though.

File: source/core/src/main/com/csse3200/game/services/DialogueBoxService.java (hunk 1 above)
Comment (LucasRyall) [reply source/core/src/main/com/csse3200/game/services/DialogueBoxService.java:89]: My sonarlint (which is synced with our project's sonarcloud) has no issue with it.

**Review by PHONGNGYUEN1 — APPROVED** [review]
all good to go

**Pull request merged**